### PR TITLE
Add Scheduler ResourceManager Powershell.

### PR DESCRIPTION
**Pull request by Sumanan**
```
     This change add following cmdlets for RM Powershell:

     1) Disable-AzureRmSchedulerJobCollection
     2)  Enable-AzureRmSchedulerJobCollection
     3)  Get-AzureRmSchedulerJobCollection
     4)  New-AzureRmSchedulerJobCollection
     5)  Remove-AzureRmSchedulerJobCollection
     6)  Set-AzureRmSchedulerJobCollection
     7)  Get-AzureRmSchedulerJob
     8)  Get-AzureRmSchedulerJobHistory
     9)  New-AzureRmSchedulerHttpJob
     10) New-AzureRmSchedulerServiceBusQueueJob
     11) New-AzureRmSchedulerServiceBusTopicJob
     12) New-AzureSchedulerStorageQueueJob
     13) Remove-AzureSchedulerJob
     14) Set-AzureRmSchedulerHttpJob
     15) Set-AzureRmSchedulerServiceBusQueueJob
     16) Set-AzureRmSchedulerServiceBusTopicJob
     17) Set-AzureSchedulerStorageQueueJob

     Missing from this Change:
     Support for Advance schedule and Retry policy.
```

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Sumanan__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    <Reference Include="Microsoft.Azure.Management.Scheduler">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Scheduler.2.1.0\lib\net45\Microsoft.Azure.Management.Scheduler.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Azure.ResourceManager">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Resources.2.20.0-preview\lib\net40\Microsoft.Azure.ResourceManager.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory, Version=2.28.0.725, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.0\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj]: >2.28.0 [](start = 82, length = 6)

Microsoft.IdentityModel.Clients.ActiveDirectory should be version 2.28.1

File: src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms, Version=2.28.0.725, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.0\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj]: >Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.0 [](start = 34, length = 54)

Microsoft.IdentityModel.Clients.ActiveDirectory should be version 2.28.1

File: src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.3.2\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.3.3.2\lib\net45\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure.Authentication, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.Authentication.2.0.1-preview\lib\net45\Microsoft.Rest.ClientRuntime.Azure.Authentication.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj]: >Microsoft.Rest.ClientRuntime.Azure.Authentication.2.0.1-previe [](start = 34, length = 62)

should be version 2.2.7-preview

File: src/ResourceManager/Scheduler/Commands.Scheduler/packages.config [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+<packages>
+  <package id="Hyak.Common" version="1.0.2" targetFramework="net45" />
+  <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Resources" version="2.20.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Scheduler" version="2.1.0" targetFramework="net45" />
+  <package id="Microsoft.Bcl" version="1.1.9" targetFramework="net45" />
+  <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net45" />
+  <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
+  <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.0" targetFramework="net45" />
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler/packages.config]: >="2.28.0 [](start = 71, length = 8)

2.28.1

File: src/ResourceManager/Scheduler/Commands.Scheduler/packages.config [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  <package id="Microsoft.Net.Http" version="2.2.22" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.2" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.2" targetFramework="net45" />
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler/packages.config]: ><package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.2" targetFramework="net45" /> [](start = 2, length = 91)

please align packages.config with Commands.Scheduler.csproj. You have there reference to Microsoft.Rest.ClientRuntime.Azure.Authentication but it is not mentioned here.

File: src/ResourceManager/Scheduler/Commands.Scheduler/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler/Properties/AssemblyInfo.cs]: >ssembly: AssemblyFileVersion("1.0.0.0")] [](start = 2, length = 40)

these versions should be aligned with module version in AzureRM.Scheduler.psd1

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Scheduler.2.1.0\lib\net45\Microsoft.Azure.Management.Scheduler.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Azure.ResourceManager">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Resources.2.20.0-preview\lib\net40\Microsoft.Azure.ResourceManager.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Azure.Test.Framework">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Test.Framework.1.0.6040.17521-prerelease\lib\net45\Microsoft.Azure.Test.Framework.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj]: >.1.0.6040.17521-prerelease [](start = 64, length = 26)

should be version 1.0.6052.28118-prerelease

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+    </Reference>
+    <Reference Include="Microsoft.Azure.Test.HttpRecorder, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Test.HttpRecorder.1.6.5-preview\lib\net45\Microsoft.Azure.Test.HttpRecorder.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj]: >Microsoft.Azure.Test.HttpRecorder.1.6.5-preview [](start = 34, length = 47)

should be version 1.6.7-preview

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+    <Reference Include="Microsoft.Data.OData, Version=5.6.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Data.OData.5.6.4\lib\net40\Microsoft.Data.OData.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Data.Services.Client, Version=5.6.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Data.Services.Client.5.6.4\lib\net40\Microsoft.Data.Services.Client.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory, Version=2.28.0.725, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.0\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj]: >2.28.0 [](start = 82, length = 6)

should be version 2.28.1

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms, Version=2.28.0.725, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.0\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj]: >ory.2.28.0 [](start = 78, length = 10)

should be version 2.28.1

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.0.0\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj]: >.2.0.0 [](start = 62, length = 6)

should be version 2.3.2

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.3.0.1\lib\net45\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj]: >e.3.0.1\l [](start = 67, length = 9)

should be 3.3.2

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.TestFramework.1.3.3-preview\lib\net45\Microsoft.Rest.ClientRuntime.Azure.TestFramework.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj]: >rk.1.3.3-preview [](start = 80, length = 16)

should be 1.3.5-preview

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/packages.config [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+  <package id="Microsoft.Azure.Management.Scheduler" version="2.1.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Test.Framework" version="1.0.6040.17521-prerelease" targetFramework="net45" />
+  <package id="Microsoft.Azure.Test.HttpRecorder" version="1.6.5-preview" targetFramework="net45" />
+  <package id="Microsoft.Bcl" version="1.1.10" targetFramework="net45" />
+  <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net45" />
+  <package id="Microsoft.Bcl.Build" version="1.0.21" targetFramework="net45" />
+  <package id="Microsoft.Data.Edm" version="5.6.4" targetFramework="net45" />
+  <package id="Microsoft.Data.OData" version="5.6.4" targetFramework="net45" />
+  <package id="Microsoft.Data.Services.Client" version="5.6.4" targetFramework="net45" />
+  <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.0" targetFramework="net45" />
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/packages.config]: please align all the versions with csproj file

**Comment by hovsepm** [comment]
> function Get-AzureRmResourceGroup

Why do you need to copy AzureRM.Resources.ps1 file to your solution? 

---
Refers to: src/ResourceManager/Scheduler/Commands.Scheduler.Test/ScenarioTests/AzureRM.Resources.ps1:1 in 8514b73. [](commit_id = 8514b73c38210ba0a216d4de0ad6e47898ed382b, deletion_comment = False)

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/ScenarioTests/JobTests.cs [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestUpdateHttpJob()
+        {
+            SchedulerController.NewInstance.RunPowerShellTests("Test-UpdateHttpJob");
+        }
+
+        //[Fact]
```
Comment (hovsepm) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/ScenarioTests/JobTests.cs]: >//[Fact] [](start = 8, length = 8)

please either remove this test or uncomment and record it

File: src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj (hunk 1 above)
Comment (Sumanan) [inline src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj]: When I increment it to 2.28.1, i'm getting Assembly mismatch. I believe its due to Commands.ResourceManager.Common still referring to 2.28.0.

File: src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj (hunk 2 above)
Comment (Sumanan) [inline src/ResourceManager/Scheduler/Commands.Scheduler/Commands.Scheduler.csproj]: When I increment it to 2.28.1, i'm getting Assembly mismatch. I believe its due to Commands.ResourceManager.Common still referring to 2.28.0.

File: src/ResourceManager/Scheduler/Commands.Scheduler/packages.config (hunk 4 above)
Comment (Sumanan) [inline src/ResourceManager/Scheduler/Commands.Scheduler/packages.config]: When I increment it to 2.28.1, i'm getting Assembly mismatch. I believe its due to Commands.ResourceManager.Common still referring to 2.28.0.

File: src/ResourceManager/Scheduler/Commands.Scheduler/packages.config (hunk 5 above)
Comment (Sumanan) [inline src/ResourceManager/Scheduler/Commands.Scheduler/packages.config]: Good catch.

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj (hunk 7 above)
Comment (Sumanan) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/Commands.Scheduler.Test.csproj]: When I updated to 1.0.6052.28118-prerelease, the tests are failing due to looking for Version=1.0.6047.28041 to which other test projects are referring to. I have updated to refer to version Version=1.0.6047.28041.

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/ScenarioTests/JobTests.cs (hunk 15 above)
Comment (Sumanan) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/ScenarioTests/JobTests.cs]: Removed.

File: src/ResourceManager/Scheduler/Commands.Scheduler.Test/ScenarioTests/AzureRM.Resources.ps1 [hunk 16]
```diff
@@ -0,0 +1,151 @@
+﻿function Get-AzureRmResourceGroup
```
Comment (markcowl) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/ScenarioTests/AzureRM.Resources.ps1]: This should be a reference and not a copy

Comment (Sumanan) [inline src/ResourceManager/Scheduler/Commands.Scheduler.Test/ScenarioTests/AzureRM.Resources.ps1]: I removed AzureRm.Resources.ps1 from the project with the second commit. Not sure how it got there in first place. Thanks.

File: src/ResourceManager/Scheduler/Commands.Scheduler/packages.config (hunk 4 above)
Comment (markcowl) [inline src/ResourceManager/Scheduler/Commands.Scheduler/packages.config]: So, the dev branch just now has 2.28.1, please pull the latest and update this.  Also Microsoft.Rest.ClientRuntime.Azure.Authentication should be 2.28.1

**Comment by markcowl** [comment]
@Sumanan a couple of changes, and please add your cmdlets to tools\AzureRM\AzureRM.psd1, analogously to the other psd1 files.  Also please add your test target here: https://github.com/Azure/azure-powershell/blob/dev/AzurePowershell.Test.targets#L71
 So your tests execute.

**Comment by markcowl** [comment]
On demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1054/

LGTM once the build passes.

**Pull request merged**